1164/en.md

```diff
@@ -0,0 +1,130 @@
+
+You will have a better understanding of this concept as you read through the tutorial. It's a straightforward problem if you can understand both the segment tree and lazy propagation algorithms.
+
+### Code
+
+```
+#include<bits/stdc++.h>
+
+using namespace std;
+
+#define ll long long
+#define endl "\n"
+#define MAX 400010
+
+
+ll segSum[MAX], lazy[MAX];
+
+void clearAll() {
+    for (int i = 0; i < MAX; i++) {
+        segSum[i] = 0;
+        lazy[i] = 0;
+    }
+}
+
+
+void update(int node, ll leftTree, ll rightTree, ll left, ll right, ll value) {
+    int leftSide  = node * 2;
+    int rightSide = (node * 2) + 1;
+    if (lazy[node]) {
+        segSum[node] += ((rightTree - leftTree + 1) * lazy[node]);
+        if (leftTree != rightTree) {
+            lazy[leftSide]  += lazy[node];
+            lazy[rightSide] += lazy[node];
+        } 
+        lazy[node] = 0;
+    }
+    if (leftTree > right || rightTree < left) {
+        return;
+    }
+    if (leftTree >= left && rightTree <= right) {
+        segSum[node] += ((rightTree - leftTree + 1) * value);
+        if (leftTree != rightTree) {
+            lazy[leftSide]  += value;
+            lazy[rightSide] += value;
+        }
+        return;
+    }
+    int mid = (leftTree + rightTree) >> 1;
+    update(leftSide, leftTree, mid, left, right, value);
+    update(rightSide, mid + 1, rightTree, left, right, value);
+    segSum[node] = segSum[leftSide] + segSum[rightSide];
+}
+
+
+ll query(int node, ll lt, ll rt, ll l, ll r) {
+    int leftSide  = node * 2;
+    int rightSide = (node * 2) + 1;
+    if (lazy[node]) {
+        segSum[node] += ((rt - lt + 1) * lazy[node]);
+        if (lt != rt) {
+            lazy[leftSide]  += lazy[node];
+            lazy[rightSide] += lazy[node];
+        } 
+        lazy[node] = 0;
+    }
+    if (lt > r || rt < l) {
+        return 0;
+    }
+    if (l <= lt && rt <= r) {
+        return segSum[node];
+    }
+    int mid = (lt + rt) >> 1;
+    ll leftSum  = query(leftSide, lt, mid, l, r);
+    ll rightSum = query(rightSide, mid + 1, rt, l, r);
+    return leftSum + rightSum;   
+}
+
+
+int  main() {
+    int t, cs = 1;
+    cin >> t;
+    while (t--) {
+        int n,q;
+        cin >> n >> q;
+        clearAll();
+        cout <<"Case "<< cs++ << ":\n";
+        while(q--) {
+            int choice, left, right;
+            cin >> choice >> left >> right;
+            if (choice == 0) {
+                ll value;
+                cin >> value;
+                update(1, 0, n - 1, left, right, value);
+            }
+            else {
+                cout << query(1, 0, n - 1, left, right) <<endl;
+            }
+        }
+    }
+    return 0;
+}
+```
+## Resources:
+- [Segment Tree](https://cp-algorithms.com/data_structures/segment_tree.html)
+- [Lazy Propagation](https://www.topcoder.com/thrive/articles/range-operations-lazy-propagation)
```

Comment:
Can we please just delete this block here and add the links to the items in the Prerequisites section, as the items remain same in both sections?

Reply:
done